cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -215,9 +215,8 @@ public void deleteApp(HttpRequest request, HttpResponder responder,
 
       // Deletion of a particular application is not allowed if that application is used by an adapter
       if (adapterService.getApplicationTemplateInfo(appId) != null) {
-        responder.sendString(HttpResponseStatus.BAD_REQUEST,
-                             String.format("Cannot delete Application %s." +
-                                             " An AdapterType exists with a conflicting name.", appId));
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, String.format(
+          "Cannot delete Application %s. An ApplicationTemplate exists with a conflicting name.", appId));
```

Comment:
Is this message correct? (Although not introduced by your PR). Perhaps 'An ApplicationTemplate is using the specified application'?